bin/check-ports.rb

```diff
@@ -86,12 +86,19 @@ def run
         port
       end
     end
+
+    hosts = config[:hosts].split(',').flat_map do |host|
+      host
+    end
+
     okarray = []
-    ports.each do |port|
-      okarray << 'ok' if check_port port
+    hosts.each do |host|
+      ports.each do |port|
+        okarray << 'ok' if check_port port, host
+      end
     end
-    if okarray.size == ports.size
-      ok "All ports (#{config[:ports]}) are accessible for host #{config[:host]}"
+    if okarray.size == ports.size * hosts.size
```

Comment:
rather than checking if the number equals it might be better to save the failed results so that you can display the failed nodes on warnings or criticals.

Reply:
The issue with this is that the way this check is written currently it throws a critical after the first bad port and doesn't check any further ports

Reply:
sigh, people should return from functions not exit, only the main program In this case `run` should be responsible for exiting. I will probably fix that up post merge to have it return a status and the message rather than immediately bailing unless you want to take a crack at it.

Reply:
I could take a crack at it but maybe better in a separate PR as it is a different issue